### PR TITLE
core: FIPS

### PR DESCRIPTION
**Pull request by BeryJu**
## Details

REPLACE ME

---

## Checklist

-   [ ] Local tests pass (`ak test authentik/`)
-   [ ] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)

If applicable

-   [ ] The documentation has been updated
-   [ ] The documentation has been formatted (`make website`)

### CODE REVIEW

File: authentik/lib/avatars.py [hunk 1]
```diff
@@ -84,7 +83,9 @@
 
 def generate_colors(text: str) -> tuple[str, str]:
     """Generate colours based on `text`"""
-    color = int(md5(text.lower().encode("utf-8")).hexdigest(), 16) % 0xFFFFFF  # nosec
+    color = (
+        int(md5(text.lower().encode("utf-8"), usedforsecurity=False).hexdigest(), 16) % 0xFFFFFF
```
[the remark this answers is not in the document]
Comment (BeryJu) [reply authentik/lib/avatars.py:87]: This is simply used as a somewhat-high entropy conversion from text to a colour

File: docker-compose.yml [hunk 2]
```diff
@@ -3,7 +3,7 @@ version: "3.4"
 
 services:
   postgresql:
-    image: docker.io/library/postgres:12-alpine
```
Comment (rissson) [inline docker-compose.yml:6]: This is a major breaking change, do we want to include it here and now?

Comment (BeryJu) [reply docker-compose.yml:6]: Well, we'll need Postgres 14 or newer as 12 defaults to md5 for hashes or something, so maybe we should make fips mode a flag and not always try to enable it on startup

Comment (rissson) [reply docker-compose.yml:6]: #9727 for upgrade instructions

File: authentik/admin/api/system.py [hunk 3]
```diff
@@ -25,11 +27,13 @@ class RuntimeDict(TypedDict):
...
 
     python_version: str
-    gunicorn_version: str
     environment: str
     architecture: str
     platform: str
     uname: str
+    openssl_version: str
+    openssl_fips_mode: bool
+    authentik_version: str
```
Comment (rissson) [inline authentik/admin/api/system.py:36]: TODO: also do this for outposts

Comment (BeryJu) [reply authentik/admin/api/system.py:36]: added with [`49e836f` (#9683)](https://github.com/goauthentik/authentik/pull/9683/commits/49e836f60bf607f16157015020cbea947d0a00a4)

File: internal/outpost/ak/api.go [hunk 4]
```diff
@@ -184,9 +186,13 @@ func (a *APIController) OnRefresh() error {
...
-		"version":   constants.VERSION,
-		"buildHash": constants.BUILD("tagged"),
-		"uuid":      a.instanceUUID.String(),
+		"version":        constants.VERSION,
+		"buildHash":      constants.BUILD("tagged"),
+		"uuid":           a.instanceUUID.String(),
+		"golangVersion":  runtime.Version(),
+		"opensslEnabled": cryptobackend.OpensslEnabled,
+		"opensslVersion": cryptobackend.OpensslVersion(),
+		"fipsEnabled":    cryptobackend.FipsEnabled,
```
[the remark this answers is not in the document]
Comment (BeryJu) [reply internal/outpost/ak/api.go:195]: For RAC I just need to switch the guac base image to be fips and then it should be fine

**Review by rissson — APPROVED** [review]

**Pull request merged**